### ISSUE TITLE
importer: unify gphotos and picasa importers

### ISSUE THREAD

**Issue by bradfitz** [perkeep/perkeep#1144]
We shouldn't have two Google Photos importers, even if Google's two APIs for accessing Google Photos are both lacking in different ways.
 
We should just do both transparently as needed.

/cc @mpl

**Comment by mpl** [comment]
I'm not even sure we'll still be able to access photos through the Drive API anyway:
https://support.google.com/photos/answer/9316089

**Comment by tswast** [comment]
Both the Picasa and the Drive API photos import are broken now. I'm getting 404 errors from the Picasa API.

> In January 2019, we will deprecate the Picasa Web Albums Data API and the Picasa Slideshow Widget. Apps that integrate with the API will stop working.

https://support.google.com/picasa/answer/6383491?hl=en

Google Photos has an API now. https://developers.google.com/photos/library/reference/rest/ I hope it will work for Perkeep's needs.

**Comment by tgulacsi** [comment]
I've checked the phitos API, it can work, though the limits seems a little bit low.


But my main concern is that I don't know whether the downloaded images contains the EXIF data, or has been stripped.




I need the original phito, so ATM I use the android uploader.




On Sat, Aug 17, 2019 at 10:09 PM +0200, "Tim Swast" <notifications@github.com> wrote:












Both the Picasa and the Drive API photos import are broken now. I'm getting 404 errors from the Picasa API.



In January 2019, we will deprecate the Picasa Web Albums Data API and the Picasa Slideshow Widget. Apps that integrate with the API will stop working.



https://support.google.com/picasa/answer/6383491?hl=en


Google Photos has an API now. https://developers.google.com/photos/library/reference/rest/ I hope it will work for Perkeep's needs.

**Comment by tgulacsi** [comment]
I'm using the Perkeep Android Uploader to sync directly, ignoring Google Photos till it strips EXIF.
This dance with Chrome (which is not my browser of choice) is just ridiculous.
Absolutely doable, but Google Photos' limitations and the direction means that we should ramp up the Takeout importing and the Android client.

**Comment by mpl** [comment]
@tgulacsi yes, it is ridiculous that we have to resort to that. But at least it is working (we have an early prototype), and it is much more likely that we'll be able to maintain it than whatever API-based solution that they'll keep breaking under our feet.

**Comment by mpl** [comment]
We now have a first rough version at https://github.com/perkeep/gphotos-cdp for those of you interested.
